clr/commands.py

```diff
@@ -1,39 +1,47 @@
 from __future__ import print_function
 from importlib import import_module
-from builtins import zip
-from builtins import object
 from dataclasses import dataclass
 import inspect
+from inspect import Signature
 import sys
 import textwrap
 import types
 import difflib
+import shelve
+import os
+from collections import namedtuple
 
 import clr.config
 
 # Sentinal for args get in get_command_spec to indicate there is no default.
-NO_DEFAULT = object()
+# Because we are pickling command specs for clr cache use a random int
+# and check for equality rather than object identity.
+# TODO(michael.cusack): Move command spec to inspect.Signature and remove this.
+NO_DEFAULT = 4194921784511160246
+
+# Sorted list of command namespace keys.
+NAMESPACE_KEYS = sorted(clr.config.commands().keys() | {'system'})
```

Comment:
Readability nit (not your code, I realize). This line is trying to give a sorted list of all commands, and ensure that `system` is present (but not double counted).

I had to look up the behavior of `__or__` on the `dict_keys` type (a generator in Py3). We might refactor to make this behavior a bit more obvious, Other possible ways:

```python
NAMESPACE_KEYS = sorted({'system', *clr.config.commands.keys()})
NAMESPACE_KEYS = sorted({'system'}.union(clr.config.commands.keys()))
```

Could also just leave it as-is to minimize changes.